transaction.go

```diff
@@ -58,11 +58,42 @@ func (t *Transaction) SetScript(script []byte) *Transaction {
 }
 
 // AddArgument adds a Cadence argument to this transaction.
-func (t *Transaction) AddArgument(arg cadence.Value) *Transaction {
+func (t *Transaction) AddArgument(arg cadence.Value) error {
+	encodedArg, err := jsoncdc.Encode(arg)
+	if err != nil {
+		return fmt.Errorf("failed to encode argument: %w", err)
+	}
+
+	t.Arguments = append(t.Arguments, encodedArg)
+	return nil
+}
+
+// AddRawArgument adds a raw JSON-CDC encoded argument to this transaction.
```

Comment:
Maybe this comment shouldn't mention JSON-CDC, as it itself does not require the passed value to be encoded that way. In the future this might confuse readers/users: it sounds like that this function only accepts JSON-CDC, though in the future it would/could be used with another encoding format.